clippy_lints/src/blocks_in_conditions.rs

```diff
@@ -86,6 +86,13 @@ impl<'tcx> LateLintPass<'tcx> for BlocksInConditions {
                         if expr.span.from_expansion() || ex.span.from_expansion() {
                             return;
                         }
+
+                        // Linting should not be triggered to cases where `return` is included in the condition.
+                        // #9911
```

Comment:
Nit: I personally dislike using a issue number here (it is already present in the tests which would fail if this was removed), but there are already a few instances of it (but not many) in the sources.  This is different than using "TODO see issue #xxx" because in this case there is an action needed.

Reply:
I think it's fine to have context like this noted